Exiled.API/Features/Room.cs

```diff
@@ -73,49 +53,33 @@ public Room(string name, Transform transform, Vector3 position)
         /// <summary>
         /// Gets a <see cref="IEnumerable{T}"/> of <see cref="Door"/> in the <see cref="Room"/>.
         /// </summary>
-        public IEnumerable<Door> Doors { get; }
-
-        /// <summary>
-        /// Equality Comparer.
-        /// </summary>
-        /// <param name="lhs">Left comparer.</param>
-        /// <param name="rhs">Right comparer.</param>
-        /// <returns>If the rooms are equal.</returns>
-        public static bool operator ==(Room lhs, Room rhs) => lhs != null && lhs.Equals(rhs);
+        public IEnumerable<Door> Doors { get; private set; }
 
-        /// <summary>
-        /// Equality Comparer.
-        /// </summary>
-        /// <param name="lhs">Left comparer.</param>
-        /// <param name="rhs">Right comparer.</param>
-        /// <returns>If the rooms are not equal.</returns>
-        public static bool operator !=(Room lhs, Room rhs) => lhs != null && !lhs.Equals(rhs);
+        private FlickerableLightController FlickerableLightController { get; set; }
 
         /// <summary>
         /// Flickers the room's lights off for a duration.
         /// </summary>
         /// <param name="duration">Duration in seconds.</param>
-        public void TurnOffLights(float duration) => flickerableLightController?.ServerFlickerLights(duration);
+        public void TurnOffLights(float duration) => FlickerableLightController?.ServerFlickerLights(duration);
 
         /// <summary>
-        /// Equality Comparer.
+        /// Factory method to create and add a <see cref="Room"/> component to a Transform.
+        /// We can add parameters to be set privately here.
         /// </summary>
-        /// <param name="other">Other Room.</param>
-        /// <returns>If the rooms are equal.</returns>
-        public bool Equals(Room other) => other != null && this.Transform.gameObject == other.Transform.gameObject;
+        /// <param name="roomGameObject">The Game Object to attach the Room component to.</param>
+        /// <returns>The Room component that was instantiated onto the Game Object.</returns>
+        internal static Room CreateComponent(GameObject roomGameObject)
+        {
+            var room = roomGameObject.AddComponent<Room>();
 
-        /// <summary>
-        /// Equality Comparer.
-        /// </summary>
-        /// <param name="obj">Other object.</param>
-        /// <returns>If the rooms are equal.</returns>
-        public override bool Equals(object obj) => obj is Room room && this == room;
+            room.Zone = FindZone(roomGameObject);
+            room.Type = FindType(roomGameObject.name);
+            room.Doors = FindDoors(roomGameObject);
+            room.FlickerableLightController = roomGameObject.GetComponentInChildren<FlickerableLightController>();
```

Comment:
Why not just use the `Awake()` method since it's a component? I.e. move these actions to method

Reply:
This is a common technique in instantiating components where you want to pass them parameters (Just like a constructor), but don't want those parameters to be public.
Right now it's not important, but in case we ever want to pass private data to the room when we instantiate it, we can use the static factor method.